src/parser/esprima_cpp/esprima.cpp

```diff
@@ -5406,15 +5449,55 @@ class Parser {
             isGenerator = true;
         }
 
-        if (kind == ClassElementNode::Kind::None && keyNode && this->match(LeftParenthesis)) {
-            kind = ClassElementNode::Kind::Method;
-            bool allowSuperCall = false;
-            if (hasSuperClass) {
-                if (keyNode && builder.isPropertyKey(keyNode, "constructor")) {
-                    allowSuperCall = true;
+        if (kind == ClassElementNode::Kind::None && keyNode) {
+            if (this->match(LeftParenthesis)) {
+                kind = ClassElementNode::Kind::Method;
+                bool allowSuperCall = false;
+                if (hasSuperClass) {
+                    if (keyNode && builder.isPropertyKey(keyNode, "constructor")) {
+                        allowSuperCall = true;
+                    }
+                }
+                value = this->parsePropertyMethodFunction(builder, allowSuperCall, isGenerator, isAsync, mayMethodStartNode);
+            } else if (this->match(Substitution)) {
+                kind = isStatic ? ClassElementNode::Kind::StaticField : ClassElementNode::Kind::Field;
+                value = this->parsePropertyInitializer(builder);
+            } else if (this->match(SemiColon)) {
+                auto metaNode = this->createNode();
+                this->nextToken();
+
+                kind = isStatic ? ClassElementNode::Kind::StaticField : ClassElementNode::Kind::Field;
+                value = this->finalize(metaNode, builder.createLiteralNode(Value()));
+            } else {
+                bool hasLineBreak = false;
+                size_t idx = this->lookahead.start - 1;
+                while (true) {
+                    auto ch = this->scanner->sourceCharAt(idx);
+                    if (EscargotLexer::isLineTerminator(ch)) {
+                        hasLineBreak = true;
+                        break;
+                    }
+                    if (!EscargotLexer::isWhiteSpace(ch)) {
+                        break;
+                    }
+                    idx--;
+                }
```

Comment:
I'm wondering why this part is inserted.
Could you explain it more?

Reply:
if there is no semicolon like
```
class C {
static a
static b = 1;
}
```
we should process even there is no semicolon like there is a semicolon

Reply:
I see.
But what if there is a comment between no-semicolon static field and next field like below?
Is it still illegal expression or do we have to deal with this case too?
```js
class C {
  static a
  /* comment */ static b = 1;
}
```

Reply:
You are right. I would consider that case. I will change this pr

Reply:
I updated pr

Reply:
Looks great!